persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -279,6 +282,12 @@ public boolean isInSchemaAgreement() {
         .build();
   }
 
+  @Override
+  public void executeAuthResponse(Runnable handler) {
```

Comment:
I think that we could achieve a cleaner and more caller friendly API by returning:
`CompletableFuture<Void>` from this method. If we do that, the calling code doesn't need to create a `CompletableFuture` itself and a callback that fulfills the CompletableFuture. wdyt?

Reply:
I might be misunderstanding what you're suggesting.

`AuthResponse.execute()` expects a return type of `CompletableFuture<? extends Response>` so we couldn't return the `CompletableFuture<Void>` directly so it'd have to call `thenApply()` or `thenApplyAsync()`. The problem with `thenApply()` is it could be called synchronously on the thread calling `thenApply()` if the future has already been completed. `thenApplyAsync()` can't be called because we con't have access to the request executor.